clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/gvcf/TransferGvcfV1Metadata.scala

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.clio.transfer.model.gvcf
+
+import java.time.OffsetDateTime
+
+import org.broadinstitute.clio.transfer.model.TransferMetadata
+import org.broadinstitute.clio.util.model.DocumentStatus
+
+case class TransferGvcfV1Metadata(analysisDate: Option[OffsetDateTime] = None,
+                                  contamination: Option[Float] = None,
+                                  documentStatus: Option[DocumentStatus] = None,
+                                  gvcfMd5: Option[String] = None,
+                                  gvcfPath: Option[String] = None,
+                                  gvcfSize: Option[Long] = None,
+                                  gvcfIndexPath: Option[String] = None,
+                                  gvcfMetricsPath: Option[String] = None,
+                                  pipelineVersion: Option[String] = None,
+                                  notes: Option[String] = None)
+    extends TransferMetadata[TransferGvcfV1Metadata] {
+
+  override def pathsToMove: Seq[String] =
+    Seq.concat(gvcfPath, gvcfIndexPath, gvcfMetricsPath)
+
+  override def rebaseOnto(destination: String): TransferGvcfV1Metadata = {
+    assert(destination.endsWith("/"))
+
+    this.copy(
```

Comment:
Why not implement using `pathsToMove`? Then you could do it in the base class.

Reply:
`pathsToMove` only gives you the path values, not the names of the parameters those values are coming from. There's no way (that I know of) for the code in the base class to do a `this.copy` and know the field names of the implementer to use in the kwargs.

Reply:
Oh right, I sometimes forget we're not storing our map-like data in an actual map.